src/autogluon_dashboard/scripts/widget.py

```diff
@@ -4,6 +4,63 @@
 
 
 class Widget:
+    """
+    This class is used to define any Panel widgets that are used on the dashboard website.
+    These widgets can be independent or attached to a dashboard to control the plot.
+    ...
+
+    Attributes
+    ----------
+    widget_type: str
+        the widget type is used to decide what type of panel widget will be created [refer to the if-else statements in __init__]
+    name: str
+        name of the widget on the website
+    value: (int/str)
+        for applicable widgets, this is the value that will be displayed [refer to number and slider widgets]
+    options: (str/List[str])
+        for applicable widgets, this is the list of options that will be displayed [refer to select and toggle widgets]
+    format: str
+        for applicable widgets, this is the format of the "value" being displayed [refer to the number indicator widget]
+    start: int
+        for applicable widgets, this is the start of a range being displayed [refer to the slider widget]
+    end: int
+        for applicable widgets, this is the end of a range being displayed [refer to the slider widget]
+    file: str
+        for applicable widgets, this is a file object to upload/download
+    filename: str
+        for applicable widgets, this is the name of the file to upload/download
+
+    Methods
+    -------
+    _create_selectwidget():
+        Creates a panel `Select` widget.
+        The `Select` widget allows selecting a `value` from a list of `options`.
+
+    _create_togglewidget():
+        Creates a panel `ToggleGroup` widget.
+        The `ToggleGroup` is a group of widgets which can be switched 'on' or 'off'. It can be displayed as buttons or boxes (default).
+
+    _create_numberwidget():
+        Creates a panel `Number` indicator.
+        The `Number` indicator renders the `value` as text.
+
+    _create_sliderwidget():
+        Creates a panel `IntSlider` widget.
+        The `IntSlider` widget allows selecting an integer value within a set of bounds using a slider.
+
+    _create_downloadwidget():
+        Creates a panel `FileDownload` widget.
+        The `FileDownload` widget allows a user to download a file.
+        It works either by sending the file data to the browser on initialization (`embed`=True), or when the button is clicked.
```

Comment:
Maybe just having a `create` method for the widget class, and subclass will implement the `create` method, i.e. `SelectWidget.create()` will do what `_create_selectwidget()` is currently doing

Reply:
Yeah sure we could do that